server/src/main/java/org/spine3/server/event/EventBus.java

```diff
@@ -268,18 +270,28 @@ public void run() {
      * The message also must satisfy validation constraints defined in its Protobuf type.
      *
      * @param event the event message to check
-     * @return an appropriate response: `OK`, `unsupported` or `invalid` event
+     * @param responseObserver the observer to obtain the result of the call;
+     *          {@link StreamObserver#onError(Throwable)} is called if an event is unsupported or invalid
+     * @return {@code true} if event is supported and valid and can be posted, {@code false} otherwise
      */
-    public Response validate(Message event) {
+    public boolean validate(Message event, StreamObserver<Response> responseObserver) {
         final EventClass eventClass = EventClass.of(event);
         if (isUnsupportedEvent(eventClass)) {
-            return unsupportedEvent(event);
+            final UnsupportedEventException unsupportedEvent = new UnsupportedEventException(event);
+            // TODO:2016-06-09:alexander.litus: consider storing an event with `Error` status
```

Comment:
No need to store events with errors, at least now. We store commands with error status because they come from outside, and we need to track this history. For EventBus we control all the code and can detect the error in the posting size after we call `onError()`.

Please remove TODOs related to your suggestion.